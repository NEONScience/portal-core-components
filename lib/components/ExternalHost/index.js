"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function () {
    return _ExternalHost.default;
  }
});
var _ExternalHost = _interopRequireDefault(require("./ExternalHost"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }