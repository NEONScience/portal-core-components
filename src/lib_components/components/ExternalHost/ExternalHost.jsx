import React from 'react';

import Link from '@material-ui/core/Link';

const HOST_TYPES = {
  ADDITIONAL_DATA: 'ADDITIONAL_DATA', // NEON and host offer different / complementary data
  REFORMATTED_DATA: 'REFORMATTED_DATA', // NEON and host offer same data in different formats
  EXCLUSIVE_DATA: 'EXCLUSIVE_DATA', // NEON does not offer any data, only host has it accessible
};

const LINK_TYPES = {
  BY_SITE: 'BY_SITE', // Show links to external host site page(s)
  BY_PRODUCT: 'BY_PRODUCT', // Show links to external host product page(s)
};

// JZ: Add DATA_SOURCE_TYPE to public API and remove these
// CPC: Since pulling this over from browse-refactor I've expanded it to
//      model additional info on _how_ some extrnal products link out.
//      API updates ideally should model all pertinent information.
const externalProducts = {
  'DP1.00001.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00002.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00003.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00004.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00005.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00006.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00007.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00014.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00022.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00023.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00024.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00033.001': {
    host: 'PHENOCAM',
  },
  'DP1.00034.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00035.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00040.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00041.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00042.001': {
    host: 'PHENOCAM',
  },
  'DP1.00043.001': {
    host: 'AERONET',
  },
  'DP1.00066.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00094.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00098.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00099.001': {
    host: 'AMERIFLUX',
  },
  'DP1.00100.001': {
    host: 'AMERIFLUX',
  },
  'DP1.10020.001': {
    host: 'BOLD',
    searches: [
      { query: 'BETN', title: 'Ground beetle sequences DNA barcode' },
      { query: 'BETP', title: 'Ground beetle sequences DNA barcode (Prototype Data)' },
    ],
  },
  'DP1.10038.001': {
    host: 'BOLD',
    searches: [
      { query: 'MOSN', title: 'Mosquito sequences DNA barcode' },
      { query: 'MOSP', title: 'Mosquito sequences DNA barcode (Prototype Data)' },
    ],
  },
  'DP1.10055.001': {
    host: 'NPN',
  },
  'DP1.10076.001': {
    host: 'BOLD',
    searches: [
      { query: 'MAMN', title: 'Small mammal sequences DNA barcode' },
      { query: 'MAMPR', title: 'Small mammal sequences DNA barcode (Prototype Data)' },
    ],
  },
  'DP1.20002.001': {
    host: 'PHENOCAM',
  },
  'DP1.20105.001': {
    host: 'BOLD',
    searches: [
      { query: 'FSHN', title: 'Fish sequences DNA barcode' },
    ],
  },
  'DP2.00006.001': {
    host: 'AMERIFLUX',
    allowNoAvailability: true,
  },
  'DP2.00020.001': {
    host: 'AMERIFLUX',
    allowNoAvailability: true,
  },
  'DP2.00023.001': {
    host: 'AMERIFLUX',
    allowNoAvailability: true,
  },
  'DP4.00002.001': {
    host: 'AMERIFLUX',
  },
  'DP4.00007.001': {
    host: 'AMERIFLUX',
  },
  'DP4.00067.001': {
    host: 'AMERIFLUX',
  },
  'DP4.00137.001': {
    host: 'AMERIFLUX',
  },
  'DP4.00200.001': {
    host: 'AMERIFLUX',
  },
  'DP4.00201.001': {
    host: 'AMERIFLUX',
  },
};

const renderExternalHostLink = (href = '', text = '', host = '', productCode = 'n/a') => (
  <Link
    href={href}
    target="_blank"
    data-gtm-external-host={host}
    data-gtm-product-code={productCode}
    rel="noopener noreferrer"
  >
    {text}
  </Link>
);

const externalHosts = {
  AERONET: {
    id: 'AERONET',
    name: 'AERONET',
    projectTitle: 'AERONET (the AErosol RObotic NETwork)',
    url: 'https://aeronet.gsfc.nasa.gov',
    hostType: HOST_TYPES.EXCLUSIVE_DATA,
    linkType: LINK_TYPES.BY_SITE,
    getSiteLink: (allSites = {}, siteCode = '', productCode = 'n/a') => {
      const hrefBase = 'https://aeronet.gsfc.nasa.gov/cgi-bin/data_display_aod_v3?site=';
      const nonStandardSites = {
        BART: 'NEON_Bartlett',
        CPER: 'NEON-CPER',
        DSNY: 'NEON-Disney',
        HARV: 'NEON_Harvard',
        SOAP: 'NEON-SoaprootSaddle',
        STER: 'NEON_Sterling',
      };
      const hrefSite = Object.keys(nonStandardSites).includes(siteCode)
        ? nonStandardSites[siteCode]
        : `NEON_${siteCode}`;
      const description = Object.keys(allSites).length
        ? ` - ${allSites[siteCode].description}`
        : '';
      const text = `${siteCode}${description}`;
      return renderExternalHostLink(`${hrefBase}${hrefSite}`, text, 'AERONET', productCode);
    },
  },
  AMERIFLUX: {
    id: 'AMERIFLUX',
    name: 'AmeriFlux',
    projectTitle: 'AmeriFlux',
    url: 'https://ameriflux.lbl.gov',
    hostType: HOST_TYPES.REFORMATTED_DATA,
    linkType: LINK_TYPES.BY_SITE,
    getSiteLink: (allSites = {}, siteCode = '', productCode = 'n/a') => {
      const siteCodeMapping = {
        GUAN: 'PR-xGU',
        LAJA: 'PR-xLA',
        ABBY: 'US-xAB',
        OAES: 'US-xAE',
        BARR: 'US-xBA',
        BLAN: 'US-xBL',
        BONA: 'US-xBN',
        BART: 'US-xBR',
        CLBJ: 'US-xCL',
        CPER: 'US-xCP',
        DCFS: 'US-xDC',
        DEJU: 'US-xDJ',
        DELA: 'US-xDL',
        DSNY: 'US-xDS',
        GRSM: 'US-xGR',
        HARV: 'US-xHA',
        HEAL: 'US-xHE',
        JERC: 'US-xJE',
        JORN: 'US-xJR',
        KONA: 'US-xKA',
        KONZ: 'US-xKZ',
        LENO: 'US-xLE',
        MOAB: 'US-xMB',
        MLBS: 'US-xML',
        NOGP: 'US-xNG',
        ONAQ: 'US-xNQ',
        NIWO: 'US-xNW',
        PUUM: 'US-xPU',
        RMNP: 'US-xRM',
        ORNL: 'US-xRN',
        OSBS: 'US-xSB',
        SCBI: 'US-xSC',
        SERC: 'US-xSE',
        STER: 'US-xSL',
        SOAP: 'US-xSP',
        SRER: 'US-xSR',
        STEI: 'US-xST',
        TALL: 'US-xTA',
        TEAK: 'US-xTE',
        TOOL: 'US-xTL',
        TREE: 'US-xTR',
        UKFS: 'US-xUK',
        UNDE: 'US-xUN',
        WOOD: 'US-xWD',
        WREF: 'US-xWR',
        YELL: 'US-xYE',
      };
      if (!siteCodeMapping[siteCode]) { return null; }
      const hrefBase = 'https://ameriflux.lbl.gov/sites/siteinfo';
      const hrefSite = siteCodeMapping[siteCode];
      const description = Object.keys(allSites).length
        ? ` - ${allSites[siteCode].description}`
        : '';
      const text = `${siteCode} - ${description}`;
      return renderExternalHostLink(`${hrefBase}/${hrefSite}/`, text, 'AMERIFLUX', productCode);
    },
  },
  BOLD: {
    id: 'BOLD',
    name: 'BOLD',
    projectTitle: 'BOLD (Barcode of Life Datasystem)',
    url: 'https://boldsystems.org/',
    hostType: HOST_TYPES.ADDITIONAL_DATA,
    linkType: LINK_TYPES.BY_PRODUCT,
    hostDataVariety: 'Raw sequence data',
    getProductLinks: (productCode = '') => {
      if (!externalProducts[productCode]) { return []; }
      return externalProducts[productCode].searches.map((search) => ({
        key: search.query,
        node: renderExternalHostLink(
          `https://portal.boldsystems.org/recordset/${search.query}`,
          search.title,
          'BOLD',
          productCode,
        ),
      }));
    },
  },
  NPN: {
    id: 'NPN',
    name: 'USA-NPN',
    projectTitle: 'USA National Phenology Network (USA-NPN)',
    url: 'https://www.usanpn.org/usa-national-phenology-network',
    hostType: HOST_TYPES.REFORMATTED_DATA,
    linkType: null,
  },
  PHENOCAM: {
    id: 'PHENOCAM',
    name: 'PhenoCam',
    projectTitle: 'The PhenoCam Network',
    url: 'https://phenocam.nau.edu/webcam/about/',
    hostType: HOST_TYPES.EXCLUSIVE_DATA,
    linkType: LINK_TYPES.BY_SITE,
    getSiteLink: (allSites = {}, siteCode = '', productCode = '') => {
      if (!allSites[siteCode]) { return null; }
      const hrefBase = 'https://phenocam.nau.edu/webcam/sites';
      const hrefSite = `${allSites[siteCode].domainCode}.${siteCode}.${productCode.split('.').slice(0, 2).join('.')}`;
      const text = `${siteCode} - ${allSites[siteCode].description}`;
      return renderExternalHostLink(`${hrefBase}/NEON.${hrefSite}/`, text, 'PHENOCAM', productCode);
    },
  },
};
Object.keys(externalHosts).forEach((hostId) => {
  externalHosts[hostId].renderLink = (productCode = 'n/a') => renderExternalHostLink(
    externalHosts[hostId].url,
    externalHosts[hostId].projectTitle,
    hostId,
    productCode,
  );
  externalHosts[hostId].renderShortLink = (productCode = 'n/a') => renderExternalHostLink(
    externalHosts[hostId].url,
    externalHosts[hostId].name,
    hostId,
    productCode,
  );
});

const getByHostId = (hostId = '') => (
  externalHosts[hostId] || null
);

const getByProductCode = (productCode = '') => (
  externalProducts[productCode] ? externalHosts[externalProducts[productCode].host] : null
);

const getProductSpecificInfo = (productCode = '') => (
  externalProducts[productCode] ? externalProducts[productCode] : null
);

const ExternalHost = {
  HOST_TYPES,
  LINK_TYPES,
  getByHostId,
  getByProductCode,
  getProductSpecificInfo,
  renderExternalHostLink,
};

export default ExternalHost;
