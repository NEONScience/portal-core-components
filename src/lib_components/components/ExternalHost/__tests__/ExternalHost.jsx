import React from 'react';
import renderer from 'react-test-renderer';

import ExternalHost from '../ExternalHost';
import sitesJSON from '../../../../sampleData/sites.json';

const { getByHostId, getByProductCode } = ExternalHost;

describe('ExternalHost', () => {
  test('getByProductCode and getByHostId return same host for congruent inputs', () => {
    expect(getByProductCode('DP1.00001.001')).toStrictEqual(getByHostId('AMERIFLUX'));
    expect(getByProductCode('DP1.00042.001')).toStrictEqual(getByHostId('PHENOCAM'));
    expect(getByProductCode('DP1.10038.001')).toStrictEqual(getByHostId('BOLD'));
    expect(getByProductCode('NOT VALID')).toStrictEqual(getByHostId('NOT VALID'));
  });

  /**
     AERONET
  */
  describe('AERONET', () => {
    test('getSiteLink with no args', () => {
      const tree = renderer
        .create(getByHostId('AERONET').getSiteLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('getSiteLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('AERONET').getSiteLink(sitesJSON, 'ABBY'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('getSiteLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('AERONET').getSiteLink(sitesJSON, 'ABBY', 'DP1.00043.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('AERONET').renderLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('AERONET').renderLink('DP1.00043.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('AERONET').renderShortLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('AERONET').renderShortLink('DP1.00043.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  /**
     AMERIFLUX
  */
  describe('AMERIFLUX', () => {
    test('getSiteLink with no args', () => {
      const tree = renderer
        .create(getByHostId('AMERIFLUX').getSiteLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('getSiteLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('AMERIFLUX').getSiteLink(sitesJSON, 'ABBY'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('getSiteLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('AMERIFLUX').getSiteLink(sitesJSON, 'ABBY', 'DP1.00001.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('AMERIFLUX').renderLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('AMERIFLUX').renderLink('DP1.00001.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('AMERIFLUX').renderShortLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('AMERIFLUX').renderShortLink('DP1.00001.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  /**
     PHENOCAM
  */
  describe('PHENOCAM', () => {
    test('getSiteLink with no args', () => {
      const tree = renderer
        .create(getByHostId('PHENOCAM').getSiteLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('getSiteLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('PHENOCAM').getSiteLink(sitesJSON, 'ABBY'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('getSiteLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('PHENOCAM').getSiteLink(sitesJSON, 'ABBY', 'DP1.00033.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('PHENOCAM').renderLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('PHENOCAM').renderLink('DP1.00033.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('PHENOCAM').renderShortLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('PHENOCAM').renderShortLink('DP1.00033.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  /**
     BOLD
  */
  describe('BOLD', () => {
    test('getProductLinks with no arg', () => {
      const tree = renderer
        .create(getByHostId('BOLD').getProductLinks())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('getProductLinks with a valid productCode', () => {
      const tree = renderer
        .create((
          <div>
            {getByHostId('BOLD').getProductLinks('DP1.10020.001').map((link) => (
              <div key={link.key}>{link.node}</div>
            ))}
          </div>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('BOLD').renderLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('BOLD').renderLink('DP1.10020.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('BOLD').renderShortLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('BOLD').renderShortLink('DP1.10020.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  /**
     NPN
  */
  describe('NPN', () => {
    test('renderLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('NPN').renderLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('NPN').renderLink('DP1.10055.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink without productCode', () => {
      const tree = renderer
        .create(getByHostId('NPN').renderShortLink())
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('renderShortLink with productCode', () => {
      const tree = renderer
        .create(getByHostId('NPN').renderShortLink('DP1.10055.001'))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
