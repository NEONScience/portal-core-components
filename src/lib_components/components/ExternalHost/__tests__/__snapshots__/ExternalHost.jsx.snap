// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalHost AERONET getSiteLink with no args 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AERONET"
  data-gtm-product-code="n/a"
  href="https://aeronet.gsfc.nasa.gov/cgi-bin/data_display_aod_v3?site=NEON_"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  
</a>
`;

exports[`ExternalHost AERONET getSiteLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AERONET"
  data-gtm-product-code="DP1.00043.001"
  href="https://aeronet.gsfc.nasa.gov/cgi-bin/data_display_aod_v3?site=NEON_ABBY"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  ABBY - Abby Road
</a>
`;

exports[`ExternalHost AERONET getSiteLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AERONET"
  data-gtm-product-code="n/a"
  href="https://aeronet.gsfc.nasa.gov/cgi-bin/data_display_aod_v3?site=NEON_ABBY"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  ABBY - Abby Road
</a>
`;

exports[`ExternalHost AERONET renderLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AERONET"
  data-gtm-product-code="DP1.00043.001"
  href="https://aeronet.gsfc.nasa.gov"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  AERONET (the AErosol RObotic NETwork)
</a>
`;

exports[`ExternalHost AERONET renderLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AERONET"
  data-gtm-product-code="n/a"
  href="https://aeronet.gsfc.nasa.gov"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  AERONET (the AErosol RObotic NETwork)
</a>
`;

exports[`ExternalHost AERONET renderShortLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AERONET"
  data-gtm-product-code="DP1.00043.001"
  href="https://aeronet.gsfc.nasa.gov"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  AERONET
</a>
`;

exports[`ExternalHost AERONET renderShortLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AERONET"
  data-gtm-product-code="n/a"
  href="https://aeronet.gsfc.nasa.gov"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  AERONET
</a>
`;

exports[`ExternalHost AMERIFLUX getSiteLink with no args 1`] = `null`;

exports[`ExternalHost AMERIFLUX getSiteLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AMERIFLUX"
  data-gtm-product-code="DP1.00001.001"
  href="https://ameriflux.lbl.gov/sites/siteinfo/US-xAB/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  ABBY -  - Abby Road
</a>
`;

exports[`ExternalHost AMERIFLUX getSiteLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AMERIFLUX"
  data-gtm-product-code="n/a"
  href="https://ameriflux.lbl.gov/sites/siteinfo/US-xAB/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  ABBY -  - Abby Road
</a>
`;

exports[`ExternalHost AMERIFLUX renderLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AMERIFLUX"
  data-gtm-product-code="DP1.00001.001"
  href="https://ameriflux.lbl.gov"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  AmeriFlux
</a>
`;

exports[`ExternalHost AMERIFLUX renderLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AMERIFLUX"
  data-gtm-product-code="n/a"
  href="https://ameriflux.lbl.gov"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  AmeriFlux
</a>
`;

exports[`ExternalHost AMERIFLUX renderShortLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AMERIFLUX"
  data-gtm-product-code="DP1.00001.001"
  href="https://ameriflux.lbl.gov"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  AmeriFlux
</a>
`;

exports[`ExternalHost AMERIFLUX renderShortLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="AMERIFLUX"
  data-gtm-product-code="n/a"
  href="https://ameriflux.lbl.gov"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  AmeriFlux
</a>
`;

exports[`ExternalHost BOLD getProductLinks with a valid productCode 1`] = `
<div>
  <div>
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      data-gtm-external-host="BOLD"
      data-gtm-product-code="DP1.10020.001"
      href="https://portal.boldsystems.org/recordset/BETN"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      Ground beetle sequences DNA barcode
    </a>
  </div>
  <div>
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      data-gtm-external-host="BOLD"
      data-gtm-product-code="DP1.10020.001"
      href="https://portal.boldsystems.org/recordset/BETP"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      Ground beetle sequences DNA barcode (Prototype Data)
    </a>
  </div>
</div>
`;

exports[`ExternalHost BOLD getProductLinks with no arg 1`] = `null`;

exports[`ExternalHost BOLD renderLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="BOLD"
  data-gtm-product-code="DP1.10020.001"
  href="https://boldsystems.org/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  BOLD (Barcode of Life Datasystem)
</a>
`;

exports[`ExternalHost BOLD renderLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="BOLD"
  data-gtm-product-code="n/a"
  href="https://boldsystems.org/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  BOLD (Barcode of Life Datasystem)
</a>
`;

exports[`ExternalHost BOLD renderShortLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="BOLD"
  data-gtm-product-code="DP1.10020.001"
  href="https://boldsystems.org/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  BOLD
</a>
`;

exports[`ExternalHost BOLD renderShortLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="BOLD"
  data-gtm-product-code="n/a"
  href="https://boldsystems.org/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  BOLD
</a>
`;

exports[`ExternalHost NPN renderLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="NPN"
  data-gtm-product-code="DP1.10055.001"
  href="https://www.usanpn.org/usa-national-phenology-network"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  USA National Phenology Network (USA-NPN)
</a>
`;

exports[`ExternalHost NPN renderLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="NPN"
  data-gtm-product-code="n/a"
  href="https://www.usanpn.org/usa-national-phenology-network"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  USA National Phenology Network (USA-NPN)
</a>
`;

exports[`ExternalHost NPN renderShortLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="NPN"
  data-gtm-product-code="DP1.10055.001"
  href="https://www.usanpn.org/usa-national-phenology-network"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  USA-NPN
</a>
`;

exports[`ExternalHost NPN renderShortLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="NPN"
  data-gtm-product-code="n/a"
  href="https://www.usanpn.org/usa-national-phenology-network"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  USA-NPN
</a>
`;

exports[`ExternalHost PHENOCAM getSiteLink with no args 1`] = `null`;

exports[`ExternalHost PHENOCAM getSiteLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="PHENOCAM"
  data-gtm-product-code="DP1.00033.001"
  href="https://phenocam.nau.edu/webcam/sites/NEON.D16.ABBY.DP1.00033/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  ABBY - Abby Road
</a>
`;

exports[`ExternalHost PHENOCAM getSiteLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="PHENOCAM"
  data-gtm-product-code=""
  href="https://phenocam.nau.edu/webcam/sites/NEON.D16.ABBY./"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  ABBY - Abby Road
</a>
`;

exports[`ExternalHost PHENOCAM renderLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="PHENOCAM"
  data-gtm-product-code="DP1.00033.001"
  href="https://phenocam.nau.edu/webcam/about/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  The PhenoCam Network
</a>
`;

exports[`ExternalHost PHENOCAM renderLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="PHENOCAM"
  data-gtm-product-code="n/a"
  href="https://phenocam.nau.edu/webcam/about/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  The PhenoCam Network
</a>
`;

exports[`ExternalHost PHENOCAM renderShortLink with productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="PHENOCAM"
  data-gtm-product-code="DP1.00033.001"
  href="https://phenocam.nau.edu/webcam/about/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  PhenoCam
</a>
`;

exports[`ExternalHost PHENOCAM renderShortLink without productCode 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  data-gtm-external-host="PHENOCAM"
  data-gtm-product-code="n/a"
  href="https://phenocam.nau.edu/webcam/about/"
  onBlur={[Function]}
  onFocus={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  PhenoCam
</a>
`;
